src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -7159,5 +7446,839 @@ void ExceptionTracker::ResetThreadAbortStatus(PTR_Thread pThread, CrawlFrame *pC
+}
+
+bool IsFilterStartOffset(EE_ILEXCEPTION_CLAUSE* pEHClause, DWORD_PTR dwHandlerStartPC)
+{
+    EECodeInfo codeInfo((PCODE)dwHandlerStartPC);
+    _ASSERTE(codeInfo.IsValid());
+
+    return pEHClause->FilterOffset == codeInfo.GetRelOffset();
+}
+
+void MakeCallbacksRelatedToHandler(
+    bool fBeforeCallingHandler,
+    Thread*                pThread,
+    MethodDesc*            pMD,
+    EE_ILEXCEPTION_CLAUSE* pEHClause,
+    DWORD_PTR              dwHandlerStartPC,
+    StackFrame             sf
+    )
+{
+    // Here we need to make an extra check for filter handlers because we could be calling the catch handler
+    // associated with a filter handler and yet the EH clause we have saved is for the filter handler.
+    BOOL fIsFilterHandler         = IsFilterHandler(pEHClause) && IsFilterStartOffset(pEHClause, dwHandlerStartPC);
+    BOOL fIsFaultOrFinallyHandler = IsFaultOrFinally(pEHClause);
+
+    if (fBeforeCallingHandler)
+    {
+        if (pMD->IsILStub())
+        {
+            return;
+        }
+
+        if (fIsFilterHandler)
+        {
+            EEToDebuggerExceptionInterfaceWrapper::ExceptionFilter(pMD, (TADDR) dwHandlerStartPC, pEHClause->FilterOffset, (BYTE*)sf.SP);
+
+            EEToProfilerExceptionInterfaceWrapper::ExceptionSearchFilterEnter(pMD);
+            ETW::ExceptionLog::ExceptionFilterBegin(pMD, (PVOID)dwHandlerStartPC);
+        }
+        else
+        {
+            EEToDebuggerExceptionInterfaceWrapper::ExceptionHandle(pMD, (TADDR) dwHandlerStartPC, pEHClause->HandlerStartPC, (BYTE*)sf.SP);
+
+            if (fIsFaultOrFinallyHandler)
+            {
+                EEToProfilerExceptionInterfaceWrapper::ExceptionUnwindFinallyEnter(pMD);
+                ETW::ExceptionLog::ExceptionFinallyBegin(pMD, (PVOID)dwHandlerStartPC);
+            }
+            else
+            {
+                EEToProfilerExceptionInterfaceWrapper::ExceptionCatcherEnter(pThread, pMD);
 
+                DACNotify::DoExceptionCatcherEnterNotification(pMD, pEHClause->HandlerStartPC);
+                ETW::ExceptionLog::ExceptionCatchBegin(pMD, (PVOID)dwHandlerStartPC);
+            }
+        }
+    }
+    else
+    {
+        if (pMD->IsILStub())
+        {
+            return;
+        }
+
+        if (fIsFilterHandler)
+        {
+            ETW::ExceptionLog::ExceptionFilterEnd();
+            EEToProfilerExceptionInterfaceWrapper::ExceptionSearchFilterLeave();
+        }
+        else
+        {
+            if (fIsFaultOrFinallyHandler)
+            {
+                ETW::ExceptionLog::ExceptionFinallyEnd();
+                EEToProfilerExceptionInterfaceWrapper::ExceptionUnwindFinallyLeave();
+            }
+            else
+            {
+                ETW::ExceptionLog::ExceptionCatchEnd();
+                ETW::ExceptionLog::ExceptionThrownEnd();
+                EEToProfilerExceptionInterfaceWrapper::ExceptionCatcherLeave();
+            }
+        }
+        // TODO: add the m_EHClauseInfo for profiler 
```

Comment:
What is this TODO about?

Reply:
Should it be a TODO-NewEH

Reply:
Right, I still plan to add a number of TODO-NewEHs to this PR. As for this TODO, the comment was related to the fact that the `ExceptionTracker::MakeCallbacksRelatedToHandler`, which is the old EH variant of this function, calls `m_EHClauseInfo.ResetInfo()` and IIRC, profiler stuff somehow relied on more stuff from the `m_EHClauseInfo` info than the stuff I've extracted from it and use in the new EH.